chainservice/chainservice.go

```diff
@@ -244,7 +244,7 @@ func New(
 	}
 	copts := []consensus.Option{
 		consensus.WithBroadcast(func(msg proto.Message) error {
-			return p2pAgent.BroadcastOutbound(p2p.WitContext(context.Background(), p2p.Context{ChainID: chain.ChainID()}), msg)
+			return p2pAgent.BroadcastOutbound(context.Background(), msg)
```

Comment:
`chain.ChainID()` is actually known when creating p2p agent at earlier time `agent = p2p.NewAgent(cfg ...)`, so save it inside p2pAgent